addons/binding/org.openhab.binding.opensprinkler/src/main/java/org/openhab/binding/opensprinkler/handler/OpenSprinklerPiHandler.java

```diff
@@ -0,0 +1,246 @@
+/**
+ * Copyright (c) 2014-2015 openHAB UG (haftungsbeschraenkt) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.opensprinkler.handler;
+
+import static org.openhab.binding.opensprinkler.OpenSprinklerBindingConstants.*;
+
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.State;
+import org.openhab.binding.opensprinkler.OpenSprinklerBindingConstants.Station;
+import org.openhab.binding.opensprinkler.config.OpenSprinklerPiConfig;
+import org.openhab.binding.opensprinkler.internal.api.OpenSprinklerApi;
+import org.openhab.binding.opensprinkler.internal.api.OpenSprinklerApiFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link OpenSprinklerPiHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Chris Graham - Initial contribution
+ */
+public class OpenSprinklerPiHandler extends BaseThingHandler {
```

Comment:
doesn't this class have quite a few things in common with OpenSprinklerHandler, so that they could share an abstract super class?